src/pages/Merchant/Register/Register.js

```diff
@@ -84,11 +86,52 @@ function Register() {
     }
   };
 
-  const handleSubmit = (event) => {
+  const handleOtpVerification = async (event) => {
     event.preventDefault();
-    if (!validateForm()) {
+    const dummyErrors = {};
+    if (!validateOtp(values.otp)) {
+      dummyErrors.otp = "Enter valid OTP";
+      setErrors(dummyErrors);
       return;
     }
+    if (values.otp)
+      if (!otpObj) return;
+      else {
+        const user = await verifyOtp(otpObj, values.otp);
+        if (user === "") return;
+        registerMerchant(registerDetails).then((res) => {
+          console.log(res);
+        });
+      }
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!validateForm()) return;
+
+    setRegisterDetails({
+      firstName: values.fname,
+      lastName: values.lname,
+      mobile: values.mobile,
+      isMerchant: true,
+      password: values.password,
+      email: values.email,
+    });
+
+    checkRegisterDetails(registerDetails).then(async (res) => {
+      if (!res) {
+        setOtpPage(false);
+        return;
+      } else {
```

Comment:
if you are returning from the function in the if condition then there is no use of defining else block. although it is not necessary but just saying

Reply:
ok